linearmodels/iv/_utility.py

```diff
@@ -78,12 +83,27 @@ class IVFormulaParser:
     The general structure of a formula is `dep ~ exog + [endog ~ instr]`
     """
 
-    def __init__(self, formula: str, data: DataFrame, eval_env: int = 2):
+    def __init__(
+        self,
+        formula: str,
+        data: DataFrame,
+        eval_env: int = 2,
+        context: LayeredMapping | None = None,
+    ):
         self._formula = formula
         self._data = data
         self._eval_env = eval_env
+        if not context:
+            self._context = capture_context(context=self._eval_env)
+        else:
+            self._context = context
         self._components: dict[str, str] = {}
         self._parse()
+        if FORMULAIC_GTE_0_6 and not future_ordering():
+            self._formulaic_kwargs = dict(_ordering="sort")
+            self._model_matrix_kwargs = dict(cluster_by="numerical_factors")
```

Comment:
Note: This was *not* the default behaviour from before. This was added in 0.6 to allow matching `patsy`'s behaviour. To match formulaic < 0.6, omit the `cluster_by` option here.

You can choose whether you want the future behaviour to cluster like patsy or not. I can see good arguments for both, but I lean to accepting the default of `cluster_by="none"`.

Reply:
Thanks. Changed.